src/com/xilinx/rapidwright/edif/ParallelEDIFParser.java

```diff
@@ -0,0 +1,283 @@
+ *  
+ * This file is part of RapidWright. 
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+ 
+package com.xilinx.rapidwright.edif;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.xilinx.rapidwright.device.Device;
+import com.xilinx.rapidwright.tests.CodePerfTracker;
+import com.xilinx.rapidwright.util.ParallelismTools;
+import com.xilinx.rapidwright.util.function.InputStreamSupplier;
+
+/**
+ * Fast EDIF Parser using parallelism
+ */
+public class ParallelEDIFParser implements AutoCloseable{
+    private static final long SIZE_PER_THREAD = EDIFTokenizer.DEFAULT_MAX_TOKEN_LENGTH * 8L;
+    protected final List<ParallelEDIFParserWorker> workers = new ArrayList<>();
+    protected final Path fileName;
+    private final long fileSize;
+    protected final InputStreamSupplier inputStreamSupplier;
+    protected final int maxTokenLength;
+    protected NameUniquifier uniquifier = NameUniquifier.concurrentUniquifier();
+
+    ParallelEDIFParser(Path fileName, long fileSize, InputStreamSupplier inputStreamSupplier, int maxTokenLength) {
+        this.fileName = fileName;
+        this.fileSize = fileSize;
+        this.inputStreamSupplier = inputStreamSupplier;
+        this.maxTokenLength = maxTokenLength;
+    }
+
+    public ParallelEDIFParser(Path fileName, long fileSize, InputStreamSupplier inputStreamSupplier) {
+        this(fileName, fileSize, inputStreamSupplier, EDIFTokenizer.DEFAULT_MAX_TOKEN_LENGTH);
+    }
+
+    public ParallelEDIFParser(Path p, long fileSize) {
+        this(p, fileSize, InputStreamSupplier.fromPath(p));
+    }
+
+    public ParallelEDIFParser(Path p) throws IOException {
+        this(p, Files.size(p));
+    }
+
+    protected ParallelEDIFParserWorker makeWorker(long offset) throws IOException {
+        return new ParallelEDIFParserWorker(fileName, inputStreamSupplier.get(), offset, uniquifier, maxTokenLength);
+    }
+
+    private int calcThreads(long fileSize) {
+        int maxUsefulThreads = Math.max((int) (fileSize / SIZE_PER_THREAD),1);
+        int processors = Runtime.getRuntime().availableProcessors();
```

Comment:
Could you replace this with a call to a new method like `ParallelismTools.maxParallelism()`?